### DISCUSSION TITLE
how to create a virtual bacnet device with properties

### DISCUSSION THREAD

**Discussion by Jzhenli** [ChristianTremblay/BAC0#486] [Q&A] [answered]
Hi, I'm newer to this lib, now I want to create a virtual bacnet device with properties, and make it can be discovered by VTS in the network, how to do it?  Would u give a simple example, thanks.

**Comment by ChristianTremblay** [comment]
Look in the test folder

manual_test_create_device.py

it creates 4 devices that talk to each others

**Reply by Jzhenli** [reply]
@ChristianTremblay Thanks for your reply, I try to modify the scripts  manual_test_create_device.py, but it seems not work, I'm not sure if I miss something, would u help to check it ?
[test.zip](https://github.com/user-attachments/files/17279965/test.zip)

**Comment by ChristianTremblay** [comment, marked as answer]
the app itself (bacnet), part of the async for loop, is the device that you create. Any BAC0 app is a BACnet device.

a BAC0.device is another BACnet device, found on the network and with which you want to interact.
